### PR TITLE
ref: isTokenAboutToExpire for clarity and maintainability

### PR DESCRIPTION
**Pull request by guipguia**
Refactor token handling function for clarity and maintainability

- Split complex logic in `isTokenAboutToExpire` into helper functions:
  - `decodePayload`: Decodes the base64 payload from the token.
  - `getExpiryTime`: Extracts and converts the expiry time from the payload.
- Simplified `isTokenAboutToExpire` main function for readability.

Solve this issue: https://github.com/headlamp-k8s/headlamp/issues/1848

### CODE REVIEW

File: backend/cmd/headlamp.go [hunk 1]
```diff
@@ -69,6 +69,8 @@ const ContextCacheTTL = 5 * time.Minute // minutes
 
 const ContextUpdateChacheTTL = 20 * time.Second // seconds
 
+const JWTExpirationTTL = 10 * time.Second //seconds
```
Comment (illume) [inline backend/cmd/headlamp.go:72]: The linter (`make backend-lint`) is asking for a space here: `// seconds`

Comment (guipguia) [reply backend/cmd/headlamp.go]: Sorry, 

I missed that, should be fixed now :)

**Review by illume — APPROVED** [review]
Looks good to me. Thanks 🎉🎈

A nitpick about the commit message... For consistency it would be good if it could start with "backend:", so something like: "`backend: istokenAboutToExpire: Refactor for clarify and maintainability`".

**Review by yolossn — APPROVED** [review]

**Comment by guipguia** [comment]
> Looks good to me. Thanks 🎉🎈
> 
> A nitpick about the commit message... For consistency it would be good if it could start with "backend:", so something like: "`backend: istokenAboutToExpire: Refactor for clarify and maintainability`".

Have updated the commit message to match the one suggested :)

**Comment by joaquimrocha** [comment]
Obrigado @guipguia !

**Pull request merged**